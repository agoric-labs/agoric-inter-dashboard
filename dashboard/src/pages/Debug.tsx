import { useEffect, useState } from 'react';
import { useCubeQuery } from '@cubejs-client/react';
import { ErrorAlert } from '@/components/ErrorAlert';
import { CubeProvider } from '@/components/CubeProvider';
import { ValueCard } from '@/components/ValueCard';
import { ValueCardGrid } from '@/components/ValueCardGrid';
import { PageHeader } from '@/components/PageHeader';
import { PageContent } from '@/components/PageContent';

function Timer() {
  const [s, setS] = useState(0);

  useEffect(() => {
    const t = setInterval(() => {
      setS((v) => v + 1);
    }, 1000);

    return () => clearInterval(t);
  }, []);

  return <ValueCard title="Last Refresh" value={`${s}s ago`} />;
}

export function Content() {
  const res = useCubeQuery(
    {
      measures: ['blocks.block_height_max'],
    },
    {
      subscribe: true,
    },
  );

  if (res.isLoading) {
    return <div>Loading...</div>;
  }

  if (res.error) {
    return <ErrorAlert value={res.error} />;
  }

  if (!res.resultSet) {
    return null;
  }

  const rows = res.resultSet.tablePivot();

  if (rows.length === 0) {
    return null;
  }

  const lastHeight = rows[0]['blocks.block_height_max'] as number;

  return (
    <>
      <PageHeader title="Debug" />
      <PageContent>
        <ValueCardGrid>
          <ValueCard title="Last Block Height" value={lastHeight} />
          <Timer key={lastHeight} />
        </ValueCardGrid>
      </PageContent>
    </>
  );
}

export function DebugPage() {
  return (
    <CubeProvider withWebsockets>
      <Content />
    </CubeProvider>
  );
}
