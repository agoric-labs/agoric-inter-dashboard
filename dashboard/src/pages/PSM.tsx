import { useLoaderData } from 'react-router-dom';
import { ResponsiveContainer, PieChart, Pie, Cell, Tooltip } from 'recharts';
import { PageHeader } from '@/components/PageHeader';
import { PageContent } from '@/components/PageContent';
import { ValueCardGrid } from '@/components/ValueCardGrid';
import { Card, CardHeader, CardTitle, CardContent } from '@/components/ui/card';
import { PSMStats } from '@/components/PSMStats';
import { BarChart } from '@/components/BarChart';
import { RadianTooltip } from '@/components/RadianTooltip';
import { colors } from '@/components/palette';

export const PSM = () => {
  const data: any = useLoaderData();
  const highStats = data.psm_stats
    .filter((s: any) => s.minted_pool_balance > 0)
    .map((s: any) => ({ ...s, utilizedPercent: s.utilized * 100 }));

  return (
    <>
      <PageHeader title="PSM" />
      <PageContent>
        <div className="grid grid-cols-1 2xl:grid-cols-2 gap-4">
          <Card>
            <CardHeader>
              <CardTitle>IST Utilization Per Anchor</CardTitle>
            </CardHeader>
            <CardContent>
              <div className="w-full h-[400px]">
                <ResponsiveContainer width="100%" height="100%">
                  <PieChart width={800} height={400}>
                    <Pie
                      dataKey="utilizedPercent"
                      data={highStats}
                      innerRadius={70}
                      outerRadius={100}
                      fill="green"
                      nameKey="label"
                      paddingAngle={3}
                      minAngle={15}
                      labelLine={false}
                      label={RadianTooltip}
                    >
                      {highStats.map((s: any, idx: number) => (
                        <Cell fill={colors[idx % colors.length]} key={s.coin} />
                      ))}
                    </Pie>
                    <Tooltip />
                  </PieChart>
                </ResponsiveContainer>
              </div>
            </CardContent>
          </Card>
          <Card>
            <CardHeader>
              <CardTitle>Total Minted IST Per Anchor</CardTitle>
            </CardHeader>
            <CardContent>
              <div className="w-full h-[400px]">
                <ResponsiveContainer width="100%" height="100%">
                  <PieChart width={800} height={400}>
                    <Pie
                      dataKey="minted_pool_balance"
                      data={highStats}
                      innerRadius={70}
                      outerRadius={100}
                      fill="green"
                      nameKey="label"
                      paddingAngle={3}
                      minAngle={15}
                      labelLine={false}
                      label={RadianTooltip}
                    >
                      {highStats.map((s: any, idx: number) => (
                        <Cell fill={colors[idx % colors.length]} key={s.coin} />
                      ))}
                    </Pie>
                    <Tooltip />
                  </PieChart>
                </ResponsiveContainer>
              </div>
            </CardContent>
          </Card>
        </div>
        <div className="mt-4" />
        <ValueCardGrid>
          {data.psm_stats.map((s: any) => (
            <PSMStats data={s} key={s.coin} />
          ))}
        </ValueCardGrid>
        <Card className="my-4 mb-4">
          <CardHeader>
            <CardTitle>Total Minted IST</CardTitle>
          </CardHeader>
          <CardContent>
            <BarChart data={data.psm_daily} xKey="date" y2Key="brand" valueKey="avg_total_minted_provided" />
          </CardContent>
        </Card>
      </PageContent>
    </>
  );
};
