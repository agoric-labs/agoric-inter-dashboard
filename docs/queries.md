## Check integrity

```sql
with mm as (
   select min(block_height) mn, max(block_height) mx
     from `xxx.raw_cosmos_hub.blocks`
)
select avg(events.count / b.event_count) as status_events
     , avg(msgs.count / b.message_count) as status_msgs
     , avg(txs.count / b.transaction_count) as status_txs
     , count(block_height) / count(distinct block_height) as status_blocks
     , count(block_height) / (max(block_height) - min(block_height)) as status_missing_total
     , sum(events.count) event_count
     , sum(msgs.count) msg_count
     , sum(txs.count) tx_count
     , count(block_height) as total_count
     , (max(block_height) - min(block_height)) as expected_total
     , (max(block_time) - min(block_time)) / count(block_height) avg_build_block_time
     , (max(_sdc_batched_at) - min(_sdc_batched_at)) / count(block_height) avg_extract_block_time
     , max(block_height) as last_block_height
     , max(block_height - prev_block_height) as max_hole
from (
  select block_height
       , block_time
       , event_count
       , message_count
       , transaction_count
       , _sdc_batched_at
       , lag(block_height) over (order by block_height) as prev_block_height
    from `xxx.raw_cosmos_hub.blocks`
    where block_height between (select mn from mm) + 500 and (select mx from mm) - 500
) b
left join (
  select block_height, count(block_height) count
    from `xxx.raw_cosmos_hub.events`
   group by 1
) events using (block_height)
left join (
  select block_height, count(block_height) count
    from `xxx.raw_cosmos_hub.messages`
   group by 1
) msgs using (block_height)
left join (
  select block_height, count(block_height) count
    from `xxx.raw_cosmos_hub.transactions`
   group by 1
) txs using (block_height)
```

## Fill missing blocks

```
kubectl port-forward agoric-mainnet-tendermint-head-5f876fb589-cvt5c 3333:3333
```

```
with mm as (
   select min(block_height) mn, max(block_height) mx
     from `xxx.raw_cosmos_hub.blocks`
)
select array_to_string(array_agg('curl -d \'' || to_json_string(to_json(struct(earliest, latest))) || '\' http://localhost:3333'), '\n'), count(earliest)
  from (
    select next as earliest
         , least(next + 99, block_height - 1) as latest
     from (
      select block_height
           , lag(block_height) over (order by block_height) as prev_block_height
        from `xxx.raw_cosmos_hub.blocks`
       where block_height between (select mn from mm) + 500 and (select mx from mm) - 500
     )
     cross join unnest(generate_array(prev_block_height + 1, block_height - 1, 100)) next
     where block_height - prev_block_height > 1
     order by 1
  )
```
