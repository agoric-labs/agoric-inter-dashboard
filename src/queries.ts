import { getDateKey, range } from "./utils";

export const PSM_DASHBOARD_QUERY = `
query {
    _metadata {
        lastProcessedHeight
    }
    psmMetrics {
        nodes {
            blockHeight
            denom
            mintedPoolBalance
        }
    }
    psmGovernances {
        nodes {
            mintLimit
            denom
        }
    }
    boardAuxes {
        nodes {
            allegedName
            decimalPlaces
        }
    }
}`;

export const PSM_GRAPH_TOKENS_QUERY = `
query {
    psmMetrics {
        nodes {
            id
            denom
        }
    }
}`;

export const PSM_TOKEN_DAILY_MINT_QUERY = (tokens: Array<string>) => `
query {
    ${tokens?.map((token: string) => `
        ${token}: psmMetricsDailies(first: 90, filter: {denom: {equalTo: "${token}"}}, orderBy:DATE_KEY_DESC) {
        nodes {
                id
                denom
                dateKey
                blockTimeLast
                totalMintedProvidedLast
            }
        }`
)}
}`
const currentDate = new Date();
export const VAULTS_DASHBOARD_QUERY = `
query {
    vaultManagerMetrics {
        nodes {
            id
            numActiveVaults
            totalCollateral
            totalDebt
            liquidatingCollateralBrand
        }
    }
    oraclePrices {
        nodes {
            priceFeedName
            typeOutAmount
            typeInAmount
        }
    }
    boardAuxes {
        nodes {
            allegedName
            decimalPlaces
        }
    }
    vaultManagerGovernances {
        nodes {
            id
            debtLimit
            liquidationMarginNumerator
            liquidationMarginDenominator
        }
    }
    vaults(filter: {state: {equalTo: "active"}}) {
        nodes {
            id
            denom
            balance
            state
            debt
            blockTime
        }
        totalCount
    }
    oraclePriceDailies(filter: {
        dateKey: { 
            in: [${getDateKey(currentDate).key}, ${getDateKey(currentDate, 1).key}] 
        }
    }) {
        nodes {
            typeInName
            typeInAmountLast
            typeOutAmountLast
            dateKey
        }
    }
}`;

export const OPEN_VAULTS_NEXT_PAGES_QUERY = (pages: number) => `
query {
    ${range(pages).map(index => `v${index}:vaults(filter: {state: {equalTo: "active"}}, first: 100, offset:${(index + 1) * 100}) {
        nodes {
            id
            denom
            balance
            state
            debt
            blockTime
        }
    }`)}
}`

export const VAULTS_DAILY_METRICS_QUERY = (tokens: string[]) => `
query {
    ${tokens?.map((token) =>
    `${token}: vaultManagerMetricsDailies(first: 90, filter:{liquidatingCollateralBrand: {equalTo: "${token}"} }, orderBy:DATE_KEY_DESC)  {
        nodes {
            id
            dateKey
            liquidatingCollateralBrand
            blockTimeLast
            totalCollateralLast
            metricsCount
            totalDebtLast
        }
    }
    ${token}_oracle: oraclePriceDailies (first: 90, filter:{typeInName: {equalTo: "${token}"}}, orderBy:DATE_KEY_DESC ) {
        nodes {
            id
            dateKey
            blockTimeLast
            typeInName
            typeInAmountLast
            typeOutAmountLast
        }
    }`)}
}`

export const RESERVE_DASHBOARD_QUERY = `
query {
    reserveMetrics {
        nodes {
            shortfallBalance
            allocations {
                nodes {
                    id
                    denom
                    value
                }
            }
        }
    }
    oraclePrices {
        nodes {
            typeInName
            typeOutAmount
            typeInAmount
        }
    }
    boardAuxes {
        nodes {
            allegedName
            decimalPlaces
        }
    }
}`


export const RESERVE_GRAPH_TOKENS_QUERY = `
query {
    reserveMetrics {
        nodes {
            allocations {
                nodes {
                    id
                    denom
                }
            }
        }
    }
}`

export const RESERVE_DAILY_METRICS_QUERY = (tokens: string[], startDate: number) => `
query {
    ${tokens?.map((token) =>
    `${token}: reserveAllocationMetricsDailies (first: 90, filter: {denom: {equalTo: "${token}"}}, orderBy:DATE_KEY_DESC) {
        nodes {
            id
            blockTimeLast
            dateKey
            valueLast
            denom
        }
    }
    ${token}_oracle: oraclePriceDailies (first: 90, filter: {typeInName: {equalTo: "${token}"}}, orderBy:DATE_KEY_DESC) {
        nodes {
          dateKey
          blockTimeLast
          typeInName
          typeInAmountLast
          typeOutAmountLast
        }
    }
    ${token}_last: reserveAllocationMetricsDailies (first: 1, filter: {denom: {equalTo: "${token}"}, valueLast: {greaterThan: "0" }, dateKey: {lessThan: ${startDate}}}, orderBy:DATE_KEY_DESC) {
        nodes {
          id
          blockTimeLast
          dateKey
          valueLast
          denom
        }
    }`)}
}`

export const INTER_DASHBOARD_QUERY = `
query {
    psmMetrics {
        nodes {
            denom
            mintedPoolBalance
            anchorPoolBalance
            mintedPoolBalance
        }
    }
    psmGovernances {
        nodes {
            denom
            mintLimit
        }
    }
    vaultManagerMetrics {
        nodes {
            id
            liquidatingCollateralBrand
            liquidatingDebtBrand
            totalDebt
            totalCollateral
        }
    }
    vaultManagerGovernances {
        nodes {
            id
            debtLimit
        }
    }
    reserveMetrics {
        nodes {
            totalFeeBurned
            shortfallBalance
            allocations {
                nodes {
                    id
                    denom
                    value
                }
            }
        }
    }
    wallets {
        totalCount
    }
    oraclePrices {
        nodes {
            id
            typeInName
            typeOutName
            typeInAmount
            typeOutAmount
            priceFeedName
        }
    }
    boardAuxes {
        nodes {
            allegedName
            decimalPlaces
        }
    }
}`

export const LIQUIDATIONS_DASHBOARD = `
query {
    boardAuxes {
        nodes {
            allegedName
            decimalPlaces
        }
    }
    vaultManagerMetrics {
        nodes {
            id
            numLiquidationsCompleted
            liquidatingCollateralValue
            numActiveVaults
            numLiquidationsCompleted
            numLiquidationsAborted
            numLiquidatingVaults
        }
    }
    vaultLiquidations (filter: {state: {equalTo: "liquidated"}}) {
        nodes {
            id
            denom
            debt
            state
            balance
            blockTime
            oraclePrice
            vaultManagerGovernance
            currentState {
                id
                denom
                debt
                state
                balance
                blockTime
            }
            liquidatingState {
                id
                denom
                debt
                state
                balance
                blockTime
            }
        }
    }
}`;

export const VAULT_STATE_DAILIES_QUERY = `
query {
    vaultStatesDailies  ( first: 90, orderBy: ID_DESC, offset: 1) {
        nodes {
            id
            blockTimeLast
            active
            closed
            liquidated
            liquidating
            liquidatedClosed
        }
    }
}`
