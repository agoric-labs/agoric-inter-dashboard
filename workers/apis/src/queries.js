import { range } from './utils';

export const graphqlQuery = {
  query: `
    {
      _metadata{
        lastProcessedHeight
        lastProcessedTimestamp
      }
      oraclePrices {
        nodes {
          typeInName
          typeOutName
          typeInAmount
          typeOutAmount
          blockTime
          blockHeight
        }
      }
      oraclePriceDailies (filter: {dateKey: { greaterThan: 20240501 } }, orderBy: DATE_KEY_DESC) {
        nodes {
          typeInName
          typeOutName
          typeInAmountLast
          typeOutAmountLast
          blockTimeLast
          blockHeightLast
        }
        totalCount
      }
      vaultManagerMetrics {
        nodes {
          id
          blockTime
          blockHeight
          totalCollateral
          totalDebt
          totalCollateralSold
          totalShortfallReceived
          liquidatingCollateralValue
          liquidatingDebtBrand
          liquidatingCollateralBrand
        }
      }
      vaultManagerGovernances  {
        nodes {
          id
          blockTime
          blockHeight
          debtLimit
          liquidationMarginNumerator
          liquidationMarginDenominator
        }
      }
      vaults (filter: {state: {equalTo: "active"}}) {
        nodes {
          id
          blockTime
          blockHeight
          denom
          balance
          state
          debt
          lockedValue
          coin
        }
        totalCount
      }
      vaultLiquidations (filter: {state: {equalTo: "liquidated"}}) {
        nodes {
            id
            blockTime
            blockHeight
            denom
            debt
            state
            balance
            currentState {
                id
                denom
                debt
                state
                balance
                blockTime
            }
            liquidatingState {
                id
                denom
                debt
                state
                balance
                blockTime
            }
        }
        totalCount
    }
    }`,
};

export const PAGINATINATED_DATA_QUERIES = (vaultsPagesCount, vaultLiquidationsPagesCount, oraclePriceDailiesPagesCount) => ({
  query: `{
    ${range(vaultsPagesCount).map(
      (index) => `vaults_${index}:vaults (filter: {state: {equalTo: "active"}}, first: 100, offset:${(index + 1) * 100}) {
      nodes {
        id
        blockTime
        blockHeight
        denom
        balance
        state
        debt
        lockedValue
        coin
      }
    }`
    )}
    ${range(vaultLiquidationsPagesCount).map(
      (index) => `vaultLiquidations_${index}:vaultLiquidations (filter: {state: {equalTo: "liquidated"}}, first: 100, offset:${
        (index + 1) * 100
      }) {
      nodes {
        id
        blockTime
        blockHeight
        denom
        debt
        state
        balance
        currentState {
            id
            denom
            debt
            state
            balance
            blockTime
        }
        liquidatingState {
            id
            denom
            debt
            state
            balance
            blockTime
        }
      }
    }`
    )}    
    ${range(oraclePriceDailiesPagesCount).map(
      (
        index
      ) => `oraclePriceDailies_${index}:oraclePriceDailies (filter:  {dateKey: { greaterThan: 20240501 } }, orderBy: DATE_KEY_DESC, first: 100, offset:${
        (index + 1) * 100
      }) {
      nodes {
        typeInName
        typeOutName
        typeInAmountLast
        typeOutAmountLast
        blockTimeLast
        blockHeightLast
      }
    }`
    )}
  }`,
});

export const LIQUIDATION_ORACLE_PRICES_DAILIES_QUERY = (tokens) => ({
  query: `
{
    ${Object.entries(tokens).map(
      ([token, oracleKeys]) =>
        `${token}:oraclePriceDailies (filter: {
        dateKey: { in: [${oracleKeys.join(', ')}] }, typeInName: {equalTo: "${token}"}
    }) {
        nodes {
            id
            typeInName
            typeOutName
            typeInAmountLast
            typeOutAmountLast
            dateKey
        }
    }`
    )}
}`,
});
