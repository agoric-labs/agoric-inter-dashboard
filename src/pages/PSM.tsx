import useSWR from 'swr';
import { AxiosError, AxiosResponse } from 'axios';
import { PageHeader } from '@/components/PageHeader';
import { PageContent } from '@/components/PageContent';
import { PSMStats } from '@/widgets/PSMStats';
import { PSMMintedPoolBalancePie } from '@/widgets/PSMMintedPoolBalancePie';
import { populateMissingDays, subQueryFetcher } from '@/utils';
import { PSM_DASHBOARD_QUERY, PSM_GRAPH_TOKENS_QUERY, PSM_TOKEN_DAILY_MINT_QUERY } from '@/queries';
import { PsmMetricsResponse, PsmMetricsDailyResponse, GraphData } from '@/types/psm-types';
import { GRAPH_DAYS } from '@/constants';

export const PSM = () => {
  const { data, error, isLoading } = useSWR<AxiosResponse, AxiosError>(PSM_DASHBOARD_QUERY, subQueryFetcher);
  const response = data?.data?.data;
  const queryData: { [key: string]: object } = {};

  response?.psmMetrics?.nodes?.forEach((node: { denom: string }) => {
    queryData[node.denom] = node;
  });
  response?.psmGovernances?.nodes?.forEach((node: { denom: string }) => {
    if (node.denom in queryData) queryData[node.denom] = { ...queryData[node.denom], ...node };
  });

  //  Queries for graph
  const { data: tokenNamesData } = useSWR<AxiosResponse, AxiosError>(PSM_GRAPH_TOKENS_QUERY, subQueryFetcher);
  const tokenNamesResponse: PsmMetricsResponse = tokenNamesData?.data.data;
  const tokenNames = tokenNamesResponse?.psmMetrics.nodes.map((node: { denom: string }) => node.denom) || [];

  const { data: dailyMetricsData, isLoading: graphDataIsLoading } = useSWR<AxiosResponse, AxiosError>(
    PSM_TOKEN_DAILY_MINT_QUERY(tokenNames),
    subQueryFetcher,
  );
  const dailyMetricsResponse: PsmMetricsDailyResponse = dailyMetricsData?.data.data;
  const graphDataMap: { [key: number]: GraphData } = {};
  Object.values(dailyMetricsResponse || {}).forEach((tokenDataList) => {
    tokenDataList?.nodes.forEach((dailyTokenMetrics) => {
      graphDataMap[dailyTokenMetrics.dateKey] = {
        ...graphDataMap[dailyTokenMetrics.dateKey],
        x: dailyTokenMetrics.blockTimeLast.split('T')[0],
        key: dailyTokenMetrics.dateKey,
        [dailyTokenMetrics.denom]: Number(dailyTokenMetrics.totalMintedProvidedLast) / 1_000_000,
      };
    });
  });
  
  const graphDataList = populateMissingDays(graphDataMap, GRAPH_DAYS);

  return (
    <>
      <PageHeader title="PSM" />
      <PageContent>
        <div className="grid gap-4 grid-cols-1 2xl:grid-cols-2">
          <div>
            <PSMStats data={queryData} error={error} isLoading={isLoading} />
          </div>
          <PSMMintedPoolBalancePie data={queryData} isLoading={isLoading} />
        </div>
        {/* <PSMHistory
          data={graphDataList}
          tokenNames={Object.keys(dailyMetricsData?.data.data || {})}
          isLoading={graphDataIsLoading}
        /> */}
      </PageContent>
    </>
  );
};
