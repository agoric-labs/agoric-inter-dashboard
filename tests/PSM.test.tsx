import React from 'react';
import useSWR from 'swr';
import { render } from '@testing-library/react';
import { PSM } from '@/pages/PSM';
import { PSM_DASHBOARD_QUERY, PSM_GRAPH_TOKENS_QUERY } from '@/queries';
import { graphDataMock, dashboardDataMock, graphNodesDataMock } from './__mocks__/PSM';

jest.mock('swr', () => {
  const originalModule = jest.requireActual('@/utils');
  return {
    ...originalModule,
    default: jest.fn(),
  };
});

global.ResizeObserver = jest.fn().mockImplementation(() => ({
  observe: jest.fn(),
  unobserve: jest.fn(),
  disconnect: jest.fn(),
}));

jest.mock('recharts', () => {
  const OriginalModule = jest.requireActual('recharts');
  return {
    ...OriginalModule,
    ResponsiveContainer: ({ children }: { children: typeof React.Children }) => (
      <OriginalModule.ResponsiveContainer width={800} height={800}>
        {children}
      </OriginalModule.ResponsiveContainer>
    ),
  };
});

describe('PSM Dashboard Snapshot tests', () => {
  it('should match snapshot when data is loaded', async () => {
    (useSWR as any).mockImplementation((query: string) => {
      const mockData =
        {
          [PSM_DASHBOARD_QUERY]: dashboardDataMock,
          [PSM_GRAPH_TOKENS_QUERY]: graphNodesDataMock,
        }[query] || graphDataMock;
      return { data: mockData, error: null, isLoading: false };
    });

    const { container } = render(<PSM />);
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot when data is loading', async () => {
    (useSWR as any).mockImplementation(() => ({ data: undefined, error: null, isLoading: true }));

    const { container } = render(<PSM />);
    expect(container).toMatchSnapshot();
  });
});
