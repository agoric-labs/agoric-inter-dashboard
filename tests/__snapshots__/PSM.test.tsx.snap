// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PSM Dashboard Snapshot tests should match snapshot when data is loaded 1`] = `
<div>
  <h1
    class="flex items-center space-x-4 lg:space-x-6 border-b p-8 text-3xl font-bold tracking-tight"
  >
    PSM
  </h1>
  <div
    class="p-8"
  >
    <div
      class="grid gap-4 grid-cols-1 2xl:grid-cols-2"
    >
      <div>
        <div
          class="grid grid-cols-1 xl:grid-cols-2 gap-4"
        >
          <div
            class="rounded-lg border bg-card text-card-foreground shadow-sm"
          >
            <div
              class="p-6 flex flex-row items-center justify-between space-y-0"
            >
              <h3
                class="text-2xl leading-none tracking-tight font-bold"
              >
                undefined
              </h3>
            </div>
            <div
              class="p-6 pt-0 flex-none md:flex flex-row items-center justify-between"
            >
              <div
                class="text-2xl tracking-tight mt-4"
              >
                <span
                  class="font-bold"
                  data-testid="psm-minted-ist-undefined"
                >
                  —
                   of
                   
                  <span
                    class="text-gray-400"
                  >
                    —
                  </span>
                </span>
                <p
                  class="text-sm text-muted-foreground"
                >
                  Minted IST / Limit
                </p>
              </div>
              <div
                class="text-2xl md:text-right mt-4"
              >
                <span
                  class="font-bold"
                >
                  NaN%
                </span>
                <p
                  class="text-sm text-muted-foreground"
                >
                  Utilized
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="rounded-lg border bg-card text-card-foreground shadow-sm"
      >
        <div
          class="flex flex-col space-y-1.5 p-6"
        >
          <h3
            class="text-2xl font-semibold leading-none tracking-tight"
          >
            Total Minted IST Per Anchor
          </h3>
        </div>
        <div
          class="p-6 pt-0"
        >
          <div
            class="w-full h-[400px]"
          >
            <div
              class="recharts-responsive-container"
              style="width: 800px; height: 800px; min-width: 0;"
            >
              <div
                class="recharts-wrapper"
                role="region"
                style="position: relative; cursor: default; width: 800px; height: 800px;"
              >
                <svg
                  class="recharts-surface"
                  cx="50%"
                  cy="50%"
                  height="800"
                  viewBox="0 0 800 800"
                  width="800"
                >
                  <title />
                  <desc />
                  <defs>
                    <clippath
                      id="recharts1-clip"
                    >
                      <rect
                        height="790"
                        width="790"
                        x="5"
                        y="5"
                      />
                    </clippath>
                  </defs>
                  <g
                    class="recharts-layer recharts-pie"
                    tabindex="0"
                  >
                    <g
                      class="recharts-layer"
                    >
                      <g
                        class="recharts-layer recharts-pie-sector"
                        tabindex="-1"
                      />
                    </g>
                  </g>
                </svg>
                <div
                  class="recharts-tooltip-wrapper"
                  role="dialog"
                  style="transform: translate(undefinedpx, undefinedpx); pointer-events: none; visibility: hidden; position: absolute; top: 0px; left: 0px;"
                  tabindex="-1"
                >
                  <div
                    class="recharts-default-tooltip"
                    style="margin: 0px; padding: 10px; background-color: rgb(255, 255, 255); border: 1px solid #ccc; white-space: nowrap;"
                  >
                    <p
                      class="recharts-tooltip-label"
                      style="margin: 0px;"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PSM Dashboard Snapshot tests should match snapshot when data is loading 1`] = `
<div>
  <h1
    class="flex items-center space-x-4 lg:space-x-6 border-b p-8 text-3xl font-bold tracking-tight"
  >
    PSM
  </h1>
  <div
    class="p-8"
  >
    <div
      class="grid gap-4 grid-cols-1 2xl:grid-cols-2"
    >
      <div>
        <div
          class="grid grid-cols-1 xl:grid-cols-2 gap-4"
        >
          <div
            class="rounded-lg border bg-card text-card-foreground shadow-sm"
          >
            <div
              class="p-6 flex flex-row items-center justify-between space-y-0"
            >
              <div
                class="animate-pulse bg-muted w-[150px] h-[24px] rounded-full"
              />
            </div>
            <div
              class="p-6 pt-0 flex-none md:flex flex-row items-center justify-between"
            >
              <div
                class="text-2xl tracking-tight mt-4"
              >
                <span
                  class="font-bold"
                  data-testid=""
                >
                  $0.00
                   of
                   
                  <span
                    class="text-gray-400"
                  >
                    $0.00
                  </span>
                </span>
                <p
                  class="text-sm text-muted-foreground"
                >
                  Minted IST / Limit
                </p>
              </div>
              <div
                class="text-2xl md:text-right mt-4"
              >
                <span
                  class="font-bold"
                >
                  0%
                </span>
                <p
                  class="text-sm text-muted-foreground"
                >
                  Utilized
                </p>
              </div>
            </div>
          </div>
          <div
            class="rounded-lg border bg-card text-card-foreground shadow-sm"
          >
            <div
              class="p-6 flex flex-row items-center justify-between space-y-0"
            >
              <div
                class="animate-pulse bg-muted w-[150px] h-[24px] rounded-full"
              />
            </div>
            <div
              class="p-6 pt-0 flex-none md:flex flex-row items-center justify-between"
            >
              <div
                class="text-2xl tracking-tight mt-4"
              >
                <span
                  class="font-bold"
                  data-testid=""
                >
                  $0.00
                   of
                   
                  <span
                    class="text-gray-400"
                  >
                    $0.00
                  </span>
                </span>
                <p
                  class="text-sm text-muted-foreground"
                >
                  Minted IST / Limit
                </p>
              </div>
              <div
                class="text-2xl md:text-right mt-4"
              >
                <span
                  class="font-bold"
                >
                  0%
                </span>
                <p
                  class="text-sm text-muted-foreground"
                >
                  Utilized
                </p>
              </div>
            </div>
          </div>
          <div
            class="rounded-lg border bg-card text-card-foreground shadow-sm"
          >
            <div
              class="p-6 flex flex-row items-center justify-between space-y-0"
            >
              <div
                class="animate-pulse bg-muted w-[150px] h-[24px] rounded-full"
              />
            </div>
            <div
              class="p-6 pt-0 flex-none md:flex flex-row items-center justify-between"
            >
              <div
                class="text-2xl tracking-tight mt-4"
              >
                <span
                  class="font-bold"
                  data-testid=""
                >
                  $0.00
                   of
                   
                  <span
                    class="text-gray-400"
                  >
                    $0.00
                  </span>
                </span>
                <p
                  class="text-sm text-muted-foreground"
                >
                  Minted IST / Limit
                </p>
              </div>
              <div
                class="text-2xl md:text-right mt-4"
              >
                <span
                  class="font-bold"
                >
                  0%
                </span>
                <p
                  class="text-sm text-muted-foreground"
                >
                  Utilized
                </p>
              </div>
            </div>
          </div>
          <div
            class="rounded-lg border bg-card text-card-foreground shadow-sm"
          >
            <div
              class="p-6 flex flex-row items-center justify-between space-y-0"
            >
              <div
                class="animate-pulse bg-muted w-[150px] h-[24px] rounded-full"
              />
            </div>
            <div
              class="p-6 pt-0 flex-none md:flex flex-row items-center justify-between"
            >
              <div
                class="text-2xl tracking-tight mt-4"
              >
                <span
                  class="font-bold"
                  data-testid=""
                >
                  $0.00
                   of
                   
                  <span
                    class="text-gray-400"
                  >
                    $0.00
                  </span>
                </span>
                <p
                  class="text-sm text-muted-foreground"
                >
                  Minted IST / Limit
                </p>
              </div>
              <div
                class="text-2xl md:text-right mt-4"
              >
                <span
                  class="font-bold"
                >
                  0%
                </span>
                <p
                  class="text-sm text-muted-foreground"
                >
                  Utilized
                </p>
              </div>
            </div>
          </div>
          <div
            class="rounded-lg border bg-card text-card-foreground shadow-sm"
          >
            <div
              class="p-6 flex flex-row items-center justify-between space-y-0"
            >
              <div
                class="animate-pulse bg-muted w-[150px] h-[24px] rounded-full"
              />
            </div>
            <div
              class="p-6 pt-0 flex-none md:flex flex-row items-center justify-between"
            >
              <div
                class="text-2xl tracking-tight mt-4"
              >
                <span
                  class="font-bold"
                  data-testid=""
                >
                  $0.00
                   of
                   
                  <span
                    class="text-gray-400"
                  >
                    $0.00
                  </span>
                </span>
                <p
                  class="text-sm text-muted-foreground"
                >
                  Minted IST / Limit
                </p>
              </div>
              <div
                class="text-2xl md:text-right mt-4"
              >
                <span
                  class="font-bold"
                >
                  0%
                </span>
                <p
                  class="text-sm text-muted-foreground"
                >
                  Utilized
                </p>
              </div>
            </div>
          </div>
          <div
            class="rounded-lg border bg-card text-card-foreground shadow-sm"
          >
            <div
              class="p-6 flex flex-row items-center justify-between space-y-0"
            >
              <div
                class="animate-pulse bg-muted w-[150px] h-[24px] rounded-full"
              />
            </div>
            <div
              class="p-6 pt-0 flex-none md:flex flex-row items-center justify-between"
            >
              <div
                class="text-2xl tracking-tight mt-4"
              >
                <span
                  class="font-bold"
                  data-testid=""
                >
                  $0.00
                   of
                   
                  <span
                    class="text-gray-400"
                  >
                    $0.00
                  </span>
                </span>
                <p
                  class="text-sm text-muted-foreground"
                >
                  Minted IST / Limit
                </p>
              </div>
              <div
                class="text-2xl md:text-right mt-4"
              >
                <span
                  class="font-bold"
                >
                  0%
                </span>
                <p
                  class="text-sm text-muted-foreground"
                >
                  Utilized
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="rounded-lg border bg-card text-card-foreground shadow-sm"
      >
        <div
          class="flex flex-col space-y-1.5 p-6"
        >
          <h3
            class="text-2xl font-semibold leading-none tracking-tight"
          >
            Total Minted IST Per Anchor
          </h3>
        </div>
        <div
          class="p-6 pt-0"
        >
          <div
            class="animate-pulse bg-muted w-full h-[20px] rounded-full mb-2"
          />
          <div
            class="animate-pulse bg-muted w-full h-[20px] rounded-full mb-2"
          />
          <div
            class="animate-pulse bg-muted w-full h-[20px] rounded-full mb-2"
          />
          <div
            class="animate-pulse bg-muted w-full h-[20px] rounded-full mb-2"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
